pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -1510,11 +1510,42 @@ def _convert_result(
     n_bits: Optional[int] = None,
     results_selection: Optional[list[tuple[str, int]]] = None,
 ) -> BackendResult:
+
+    for creg, reslist in resultdict.items():
+        if any(["-" in res for res in reslist]):
+            raise ValueError(
+                f"found negative value for creg: {creg}. \
+This could indicate a problem with the circuit submitted"
+            )
+
     if results_selection is None:
-        array_dict = {
-            creg: np.array([list(a) for a in reslist]).astype(np.uint8)
-            for creg, reslist in resultdict.items()
-        }
+        found_int_res = False
+        for creg, reslist in resultdict.items():
+            if any([re.findall("[23456789]", res) for res in reslist]):
+                found_int_res = True
```

Comment:
```suggestion
        found_int_res = any(re.findall("[23456789]", res) for reslist in resultdict.values() for res in reslist)
```
Slightly more efficient because of shortcutting.